### PR TITLE
Update GetTransaction RPC method to use Eth API

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

Remix/MetaMask need a get-tx-by-hash method. We had one, but it did not conform to the Eth RPC API.

### What changes were made as part of this PR:

Functional.

- Modifies our GetTx RPC method to match the Eth API

### CODE REVIEW

File: go/enclave/enclave.go [hunk 1]
```diff
@@ -277,29 +276,9 @@ func (e *enclaveImpl) Nonce(address gethcommon.Address) uint64 {
...
-			}
-		}
-		rollup = e.storage.ParentRollup(rollup)
-		if rollup == nil || rollup.Header.Number.Uint64() == common.L2GenesisHeight {
-			return nil
-		}
-	}
+func (e *enclaveImpl) GetTransaction(txHash gethcommon.Hash) (*common.L2Tx, gethcommon.Hash, uint64, uint64, error) {
+	// todo - joel - encrypt
+	return e.storage.GetTransaction(txHash)
```
Comment (tudor-malene) [inline go/enclave/enclave.go:281]: I had raised a ticket for this.
 I assigned you just now: https://github.com/obscuronet/obscuro-internal/issues/568

File: go/host/client_api_eth.go [hunk 2]
```diff
@@ -117,3 +132,72 @@ func extRollupToBlock(extRollup *common.ExtRollup) map[string]interface{} {
...
+	Type             hexutil.Uint64      `json:"type"`
+	Accesses         *types.AccessList   `json:"accessList,omitempty"`
+	ChainID          *hexutil.Big        `json:"chainId,omitempty"`
+	V                *hexutil.Big        `json:"v"`
+	R                *hexutil.Big        `json:"r"`
+	S                *hexutil.Big        `json:"s"`
+}
+
+// Lifted from Geth's internal `ethapi` package.
+func newRPCTransaction(tx *types.Transaction, blockHash gethcommon.Hash, blockNumber uint64, index uint64, baseFee *big.Int, signer types.Signer) *RPCTransaction {
```
Comment (tudor-malene) [inline go/host/client_api_eth.go:160]: same q. What is the diff?

**Review by tudor-malene — COMMENTED** [review]
looks good in principle.
A few questions.

File: go/host/client_api_eth.go [hunk 3]
```diff
@@ -117,3 +132,72 @@ func extRollupToBlock(extRollup *common.ExtRollup) map[string]interface{} {
 		"transactions":     extRollup.TxHashes,
 	}
 }
+
+// RPCTransaction is lifted from Geth's internal `ethapi` package.
+type RPCTransaction struct {
```
[the remark this answers is not in the document]
Comment (joeldudley) [reply go/host/client_api_eth.go:137]: Unchanged, but it's defined in `internal/ethapi` unfortunately :(

File: go/host/client_api_eth.go (hunk 2 above)
Comment (joeldudley) [reply go/host/client_api_eth.go:160]: As above

File: go/enclave/enclave.go (hunk 1 above)
Comment (joeldudley) [reply go/enclave/enclave.go:281]: Thanks

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**